detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/out/TxtOutputReport.kt

```diff
@@ -11,6 +11,8 @@ class TxtOutputReport : OutputReport() {
 
     override fun render(detektion: Detektion): String {
         val smells = detektion.findings.flatMap { it.value }
-        return smells.joinToString("\n") { it.compactWithSignature() }
+        return smells.joinToString("\n") { it.compactWithSignature() }.let {
```

Comment:
The `if else` here is redundant.
You can add a new line after every smell like this.

```suggestion
        return smells.joinToString("\n", postfix = "\n") { it.compactWithSignature() }
```

Reply:
Yes, and this is what this code does... If you mean this: `smells.joinToString { it.compactWithSignature() + "\n" }` it would work too(out was my first impregnation). But we are losing the performance of string builder. That would create 2k useless strings.

Reply:
Uh sorry. By looking at the test code once more, it becomes clear what this code is intended to do.
But something like this should work without introducing a new branch.

```suggestion
val builder = StringBuilder();
smells.forEach { builder.append(it.compactWithSignature()).append('\n') }
return builder.toString()
```

Reply:
Agree! This is easier.